pkg/hs/urbit-king/lib/Urbit/Vere/Dawn.hs

```diff
@@ -168,122 +153,114 @@ dawnPostRequests endpoint responseBuilder requests = do
   toFullRequest (rid, req) = object [ "jsonrpc" .= ("2.0" :: Text)
                                     , "method"  .= getRequestMethod req
                                     , "params"  .= req
-                                    , "id"      .= rid
+                                    , "id"      .= (show rid)
                                     ]
 
 -- Azimuth JSON Requests -------------------------------------------------------
 
--- Not a full implementation of the Ethereum ABI, but just the ability to call
--- a method by encoded id (like 0x63fa9a87 for `points(uint32)`), and a single
--- UIntN 32 parameter.
-encodeCall :: Text -> Int -> Text
-encodeCall method idx = method <> leadingZeroes <> renderedNumber
-  where
-    renderedNumber = pack $ showHex idx ""
-    leadingZeroes = replicate (64 - length renderedNumber) '0'
-
-data BlockRequest = BlockRequest
-  deriving (Show, Eq)
-
-instance RequestMethod BlockRequest where
-  getRequestMethod BlockRequest = "eth_blockNumber"
-
-instance ToJSON BlockRequest where
-  toJSON BlockRequest = Array $ fromList []
-
--- No need to parse, it's already in the format we'll pass as an argument to
--- eth calls which take a block number.
-parseBlockRequest :: BlockRequest -> Text -> TextBlockNum
-parseBlockRequest _ txt = txt
-
-type TextBlockNum = Text
-
-data PointRequest = PointRequest
-  { grqHexBlockNum :: TextBlockNum
-  , grqPointId :: Int
-  } deriving (Show, Eq)
+data PointResponse = PointResponse
+  --NOTE  also contains dominion and ownership, but not actually used here
+  { prNetwork :: PointNetwork
+  } deriving (Show, Eq, Generic)
+
+instance FromJSON PointResponse where
+  parseJSON (Object o) = do
```

Comment:
Okay so rather than pattern matching on Object and erroring if it's not an Object, suggest you use `withObject "PointResponse"` here, as you did in the RawResponse instance. This will have better error handling properties, and aeson's intended way of expressing this idea.